src/ocl/convolutionocl.cpp

```diff
@@ -1003,115 +661,18 @@ std::size_t ConvolutionDescriptor::GetForwardSolutionWorkspaceSize(Handle& handl
                      " is not applicable to the current problem");
 }
 
-// Todo: remove when all immediate mode calls will support invokers
-static std::vector<KernelInvoke> CompileSolver(const Handle& handle,
-                                               ConvolutionContext& ctx,
-                                               const ProblemDescription& problem,
-                                               solver::Id solver_id,
-                                               const FindDbKCacheKey& key)
-{
-    ctx.DetectRocm();
-    ctx.SetupFloats(problem);
-
-    const auto solver = solver_id.GetSolver();
-    auto db           = GetDb(ctx);
-    const auto solution =
-        solver.FindSolution(ctx, problem, db, {}); // auto tune is not expected here
-
-    std::vector<KernelInvoke> kernels;
-    AddKernels(handle, key.algorithm_name, key.network_config, solution, &kernels);
-    return kernels;
-}
-
-static Invoker PrepareInvoker(Handle& handle,
-                              ConvolutionContext& ctx,
-                              const ProblemDescription& problem,
-                              const NetworkConfig& config,
-                              solver::Id solver_id,
-                              conv::Direction dir)
-{
-    ctx.DetectRocm();
-    ctx.SetupFloats(problem);
-
-    const auto solver = solver_id.GetSolver();
-    auto db           = GetDb(ctx);
-    auto solution     = solver.FindSolution(ctx, problem, db, {}); // auto tune is not expected here
-    const auto invoker =
-        handle.PrepareInvoker(*solution.invoker_factory, solution.construction_params);
-
-    handle.RegisterInvoker(
-        invoker, config, solver_id.ToString(), AlgorithmName(solver_id.GetAlgo(dir)));
-    return invoker; // NOLINT (performance-no-automatic-move)
-}
-
-Invoker LoadOrPrepareInvoker(Handle& handle,
-                             ConvolutionContext& ctx,
-                             const ProblemDescription& problem,
-                             solver::Id solver_id,
-                             conv::Direction dir)
-{
-    const auto config = problem.BuildConfKey();
-    auto invoker      = handle.GetInvoker(config, solver_id);
-    if(invoker)
-        return *invoker;
-    return PrepareInvoker(handle, ctx, problem, config, solver_id, dir);
-}
-
 static bool CheckInvokerSupport(const solver::Id solver_id, conv::Direction dir)
 {
     const auto& algo = solver_id.GetAlgo(dir);
     return CheckInvokerSupport(algo);
 }
 
```

Comment:
All solvers support invokers, perhaps this and other such logic can be removed ?

Reply:
Yes, but it's been removed in a PR by @atamazov, I think. I can do it here as well if that's needed.